### PR TITLE
[SPARK-19337] [ML] [Doc] Documentation and examples for LinearSVC

### PR DESCRIPTION
**Pull request by hhbyyh**
## What changes were proposed in this pull request?

Documentation and examples (Java, scala, python, R) for LinearSVC

## How was this patch tested?
local doc generation

### CODE REVIEW

**Review by wangmiao1981 — COMMENTED** [review]
@hhbyyh In R, the LinearSVC is named as spark.svmLinear. I just created a PR #16969 for R example and vignettes. Can you drop off the R example? After merging the two PRs, I will update the document section to R. Or you can copy the R example to this PR.

**Comment by hhbyyh** [comment]
I see. I will drop the R example here, whichever PR goes in later can finish the document update.

File: docs/ml-classification-regression.md [hunk 1]
```diff
@@ -363,6 +363,51 @@ Refer to the [R API docs](api/R/spark.mlp.html) for more details.
...
 </div>
 
+## Linear Support Vector Machine
+
+A [support vector machine](https://en.wikipedia.org/wiki/Support_vector_machine) constructs a hyperplane
+or set of hyperplanes in a high- or infinite-dimensional space, which can be used for classification,
+regression, or other tasks. Intuitively, a good separation is achieved by the hyperplane that has
+the largest distance to the nearest training-data point of any class (so-called functional margin),
+since in general the larger the margin the lower the generalization error of the classifier. LinearSVC
+in Spark ML supports binary calssification with linear SVM. Internally, it optimizes the
```
Comment (felixcheung) [inline docs/ml-classification-regression.md]: calssification -> classification

File: docs/ml-classification-regression.md [hunk 2]
```diff
@@ -363,6 +363,51 @@ Refer to the [R API docs](api/R/spark.mlp.html) for more details.
 
 </div>
 
+## Linear Support Vector Machine
+
+A [support vector machine](https://en.wikipedia.org/wiki/Support_vector_machine) constructs a hyperplane
+or set of hyperplanes in a high- or infinite-dimensional space, which can be used for classification,
+regression, or other tasks. Intuitively, a good separation is achieved by the hyperplane that has
+the largest distance to the nearest training-data point of any class (so-called functional margin),
```
Comment (felixcheung) [inline docs/ml-classification-regression.md]: "largest distance" -> "longest distance"? I think?

**Comment by felixcheung** [comment]
title should say
`[SPARK-19337] [ML] [Dcoc]`
->
`[SPARK-19337] [ML] [Doc]`

File: docs/ml-classification-regression.md (hunk 2 above)
Comment (hhbyyh) [inline docs/ml-classification-regression.md]: Thanks for the comment. I think both large and long can be used to describe distance, wherever large is more suitable to describe the numeric margin. Please let me know if you have a strong preference.

File: docs/ml-classification-regression.md (hunk 1 above)
Comment (hhbyyh) [inline docs/ml-classification-regression.md]: Thanks

**Comment by hhbyyh** [comment]
Thanks for the comment @felixcheung

**Review by felixcheung — APPROVED** [review]
LGTM.
I'll leave this out for a day in case anyone else can comment.

File: docs/ml-classification-regression.md [hunk 3]
```diff
@@ -363,6 +363,44 @@ Refer to the [R API docs](api/R/spark.mlp.html) for more details.
...
 </div>
 
+## Linear Support Vector Machine
+
+A [support vector machine](https://en.wikipedia.org/wiki/Support_vector_machine) constructs a hyperplane
+or set of hyperplanes in a high- or infinite-dimensional space, which can be used for classification,
+regression, or other tasks. Intuitively, a good separation is achieved by the hyperplane that has
+the largest distance to the nearest training-data points of any class (so-called functional margin),
+since in general the larger the margin the lower the generalization error of the classifier. LinearSVC
+in Spark ML supports binomial classification with linear SVM. Internally, it optimizes the
```
Comment (felixcheung) [inline docs/ml-classification-regression.md]: actually, is there a reason you change this to `binomial classification`?

Comment (hhbyyh) [inline docs/ml-classification-regression.md]: just to be consistent with LR. But I'm not sure if it's the common expression.

Comment (felixcheung) [inline docs/ml-classification-regression.md]: do you have a link? I think binary classification is more commonly used

Comment (srowen) [inline docs/ml-classification-regression.md]: FWIW I have never head the term binomial classification and it doesn't show up in a Google search. I think it was a typo.

**Review by felixcheung — CHANGES_REQUESTED** [review]
let's change `binomial classification`

**Comment by hhbyyh** [comment]
Thanks for the review. Updated to binary. 
Also add the reference to R example.

**Review by felixcheung — APPROVED** [review]
LGTM @wangmiao1981

**Comment by felixcheung** [comment]
merged to master.

**Pull request closed without merging**